ui/js/models/resultsets_store.js

```diff
@@ -783,6 +783,13 @@ treeherder.factory('ThResultSetStore', [
         };
 
         var prependResultSets = function(repoName, data) {
+            // This request returns the treeherder git revision running on the server.
+            // If this differs from the version chosen during the UI page load, show a message.
+            if($rootScope.serverRev && $rootScope.serverRev !== data.meta.serverrev && _.isUndefined($rootScope.dismissedServerRev)) {
+                thNotify.send("Treeherder revision has changed, please reload the page", "warning", true);
+                $rootScope.dismissedServerRev = true;
+            }
```

Comment:
This is super cool @KWierso. I wonder if "The Treeherder application has changed" or "The Treeherder src code has changed" might be an improvement over "Treeherder revision has changed", in case users confuse revision with a push? Just throwing it out there.